### ISSUE TITLE
About the preference sampling

### ISSUE THREAD

**Issue by qianlin04** [baitingzbt/PEDA#3]
![图片](https://github.com/baitingzbt/PEDA/assets/48119752/c7a44c1f-be34-43ae-bb99-8ddbda4f6672)

Hello. I'm trying to understand why the preference in the dataset is not sampled from the whole preference space. I found that the sampling range depends on 'objs_normalized,' as shown here:

https://github.com/baitingzbt/PEDA/blob/d15900c962eca2009c00dd0225f018502c2523c6/data_generation/collect_env_new.py#L201C5-L201C5.

Does 'objs_normalized' represent the return vectors? If so, do they have a direct association with the 'physically plausible preference space'? 

Additionally, regarding the claim that "the preference of 100% running and 0% jumping is not achievable", I think "the preference of 100% running and 0% jumping" means that we only care about running and don't care about jumping, rather than completely prohibiting jumping. However, such preferences are excluded from the dataset. I'm not sure if the reason for excluding these preferences from the dataset is because these preferences may lead to the same policy.

Should all possible preferences be taken into account in D4MORL?

**Comment by baitingzbt** [comment]
Hi, first of all, this design is somehow due to our behavioral policy PGMORL only considered achieved rewards in each objective but does not have a `preference` as prior. In order to make sense of calculations across **different preference distribution** domains, we chose to infer the preference through L-1 normalization. For instance, `r = [1000, 1500] -> pref = [0.4, 0.6]`. Additionally, this also help to test whether PEDA can perform well when conditioned on ood preference.

To further explain, I agree that these definitions can be tricky and your point definitely also makes sense. However, there could be two problems if we do collect data conditioned on the extreme preference. First, we cannot distinguish extreme preference prior such that `[0.95, 0.5]` can have a similar performance as `[0.7, 0.3]`. We simply want to avoid this so that it doesn't confuse the model during training. Second, even if the behavioral policy can only achieve objective as extreme as `[0.7, 0.3]` in ratio, it doesn't mean something like `[0.8, 0.2]` is impossible (in fact, we do want to test whether PEDA can generalize or discover those solutions). So we decided to leave `[0.8, 0.2]` part blank for the model to figure out what it should mean.

Happy for more discussion and hope this helps!
